holoviews/plotting/bokeh/chart.py

```diff
@@ -906,18 +904,38 @@ def get_data(self, element, ranges, style):
         hover = 'hover' in self.handles
 
         # Group by stack or group dim if necessary
+        xdiff = None
+        xvals = element.dimension_values(xdim)
         if group_dim is None:
             grouped = {0: element}
+            is_dt = isdatetime(xvals)
+            try:
```

Comment:
This `try`/`except` seems like an inelegant hack. 

Please add something like `is_cat = is_categorial(xvals)` (pseudo code do not know if the function exists)